### PR TITLE
add a source for IDN Homograph Attacks

### PR DESCRIPTION
**Pull request by aptzer0x**
Creating this Pull Request from our previous conversations:

https://github.com/mitchellkrogza/phishing/pull/508

### CODE REVIEW

**Comment by glaubermagal** [comment]
@spirillen I am a little bit lazy today, not sure what I should add in the following files, so I left them empty:

- onions.csv
- wildcard.csv
- wildcard.rpz-nsdname.csv

Thoughts?

----
Follow the domains added:
`ɡithub.com`
`ɡooɡɩe.com`
`ɡooɡle.com`
`gmaiɩ.com`

Follow the domains added in punycode:
`https://xn--ithub-qmc.com`
`http://xn--ooe-8tbc2c.com`
`http://xn--oole-z7bc.com`
`https://xn--gmai-88b.com`

File: source/idn-homograph/domains.csv [hunk 1]
```diff
@@ -0,0 +1,4 @@
+ɡithub.com
```
Comment (spirillen) [inline source/idn-homograph/domains.csv:1]: Punicode... only

Each record should then have a issue, describing the records, see full example in https://github.com/mypdns/matrix/issues/1300

**Comment by spirillen** [comment]
All seems good, just thinking... isn't this more Typo Squatting? + intention? 🤔 

PS: shutting down for to day, as the powerbill are insane for the rest of the day €1 Kw/h

File: source/idn-homograph/domains.csv (hunk 1 above)
Comment (glaubermagal) [reply source/idn-homograph/domains.csv:1]: Oh, I got your point now. Ok, I will do it

**Comment by glaubermagal** [comment]
> All seems good, just thinking... isn't this more Typo Squatting? + intention? 🤔
> 
> PS: shutting down for to day, as the powerbill are insane for the rest of the day €1 Kw/h

Yea, I would say it is a type of **Typo Squatting**, where a person can register a domain using multiple characters set (Latin, Cyrillic, Greek, etc). Not sure if it is a better idea to add **IDN Homograph** as a subcategory of Typo Squatting. Need to think about it

Going to shut down for today too. See ya

File: source/idn-homograph/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+## Why IDN Homograph Attacks Are Dangerous
+
+IDN homograph attacks blend technical exploitation with social engineering, relying on both visual similarity and user trust. They are particularly dangerous because:
+
+-   They mimic familiar websites closely, lowering user suspicion.
+-   They often bypass automated detection systems, especially if users aren’t vigilant.
+
+Efforts to mitigate IDN homograph attacks include browser security updates, technical measures like script restriction in domain names, and public awareness to increase caution among users when entering personal information online.
+
+> source: https://kb.mypdns.org/articles/IDN-Homograph-Attacks
```
Comment (spirillen) [inline source/idn-homograph/README.md:39]: `https://kb.mypdns.org/articles/IDN-Homograph-Attacks` => `https://kb.mypdns.org/articles/MTX-A-88`

File: source/idn-homograph/rpz-ip.csv [hunk 3]
```diff
@@ -0,0 +1,3 @@
+15.197.148.33
```
Comment (spirillen) [inline source/idn-homograph/rpz-ip.csv:1]: Needs network cidr notation such as https://github.com/mypdns/matrix/blob/afda6ac966c683b414c25d32890f6c53cd32338e/source/spyware/rpz-ip.csv

Details for the RPZ rule: https://kb.mypdns.org/articles/MTX-A-10#the-response-ip-address-trigger-rpz-ip
RFC Draft: https://datatracker.ietf.org/doc/html/draft-vixie-dns-rpz-04#section-4.3

Comment (spirillen) [reply source/idn-homograph/rpz-ip.csv:1]: Ironic... they are fingerprinting your browser to ensure you are human...

image

**Comment by glaubermagal** [comment]
> All seems good, just thinking... isn't this more Typo Squatting? + intention? 🤔
> 
> PS: shutting down for to day, as the powerbill are insane for the rest of the day €1 Kw/h

hi @spirillen IMO, I believe it’d be better to keep IDN Homograph Attacks and Typo-Squatting into separate sections instead of making one a subsection of the other. I believe that it gives us room to expand on each if we want to add more details down the line. Just an idea. 

Thoughts?

File: source/idn-homograph/rpz-ip.csv (hunk 3 above)
Comment (spirillen) [reply source/idn-homograph/rpz-ip.csv:1]: @glaubermagal are you solving these IP addresses? then I can get this one merged

**Comment by spirillen** [comment]
issues are unsolved, consider MR as stalled, closing

**Pull request closed without merging**